### PR TITLE
Issue #7436: avoid using Locale.setDefault in tests

### PR DESCRIPTION
**Pull request by pbludov**
Issue #7436

Small step to run tests in parallel. We need to get rid of the global state change in the tests.

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/api/ScopeTest.java [hunk 1]
```diff
@@ -72,24 +71,15 @@ public void testMixedCaseSpaces() {
         assertEquals(Scope.ANONINNER, Scope.getInstance("AnonInner"), "Invalid scope");
     }
 
+    @DefaultLocale("tr_TR")
     @Test
-    public void testMixedCaseSpacesWithDifferentLocales() {
-        final Locale[] differentLocales = {new Locale("TR", "tr") };
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/api/ScopeTest.java]: TR and tr was replaced with tr_TR. Will this test both locales like before?

File: config/checkstyle_sevntu_checks.xml [hunk 2]
```diff
@@ -216,5 +216,12 @@
           <property name="methodName" value="assert(True|False)"/>
           <property name="argumentCount" value="1"/>
         </module>
+        <module name="com.github.sevntu.checkstyle.checks.coding.ForbidCertainMethodCheck">
+          <!-- Disallow Locale.setDefault in tests.
+            Use org.junitpioneer.jupiter.DefaultLocale annotation instead.
+            See https://github.com/checkstyle/checkstyle/issues/7436 -->
+          <property name="methodName" value="setDefault"/>
+          <property name="argumentCount" value="1-2"/>
```
Comment (rnveach) [inline config/checkstyle_sevntu_checks.xml]: @romani Are you ok with this?
When I originally suggested this I was under the impression we could specify the class, but this check only signifies the method name which can lead to false positives.

File: src/test/java/com/puppycrawl/tools/checkstyle/api/ScopeTest.java (hunk 1 above)
Comment (pbludov) [reply src/test/java/com/puppycrawl/tools/checkstyle/api/ScopeTest.java]: There was one item in the array: `Locale(country=TR, language=tr`, or just `tr_TR` in [BCP 47](ftp://ftp.isi.edu/in-notes/bcp/bcp47.txt) notation.

File: config/checkstyle_sevntu_checks.xml (hunk 2 above)
Comment (romani) [reply config/checkstyle_sevntu_checks.xml]: http://sevntu-checkstyle.github.io/sevntu.checkstyle/apidocs/com/github/sevntu/checkstyle/checks/coding/ForbidCertainMethodCheck.html
No type ....

It is better to not use such Check for such method name.

Comment (pbludov) [reply config/checkstyle_sevntu_checks.xml]: done

**Pull request merged**